File: aula3/Biblioteca.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;


class Livro{
private:
    string titulo;
    string autor;
    int ano;

public: 
    Livro(string t, string a, int y): titulo(t), autor(a), ano(y){}
    string getTitulo(){ return titulo;}
    string getAutor() { return autor;}
    int getAno(){return ano;}

    void exibirInfo(){
        cout << "Livor: "<< titulo 
        << " Autor: " << autor 
        << "Ano: " << ano << endl; 
    }
};

class Biblioteca{
private:
    vector<Livro> livros;
public:

    void adicionarLivro(Livro livro){
        livros.push_back(livro);
    }


    void exibirLivros(){
        cout << "Livros na Biblioteca: " << endl;
        for (Livro& livro: livros)
            livro.exibirInfo();
    }

    // Método para remover um livro pelo título
    void removerLivro(string titulo) {
    // Converter o título a ser removido para minúsculas
    transform(titulo.begin(), titulo.end(), titulo.begin(), ::tolower);

    auto it = remove_if(livros.begin(), livros.end(), [titulo](Livro& livro) {
        string tituloLivro = livro.titulo;
        // Converter o título do livro para minúsculas
        transform(tituloLivro.begin(), tituloLivro.end(), tituloLivro.begin(), ::tolower);
        return tituloLivro == titulo;
    });

    if (it != livros.end()) {
        livros.erase(it, livros.end());  // Remove o livro encontrado
        cout << "Livro \"" << titulo << "\" removido da biblioteca." << endl;
    } else {
        cout << "Livro \"" << titulo << "\" não encontrado na biblioteca." << endl;
    }
}
};

int main(){
    Livro livro1("1984", "George Orwell", 1949);
    Livro livro2("O Semhor dos Anéis", "J. R. R. Tolkien", 1954);
    Biblioteca biblioteca;
    biblioteca.adicionarLivro(livro1);
    biblioteca.exibirLivros();
    biblioteca.adicionarLivro(livro2);
    biblioteca.exibirLivros();

    cout << endl;
    biblioteca.removerLivro("1984");

    // Exibir livros novamente após remoção
    cout << endl;
    biblioteca.exibirLivros();

}

File: aula3/Biblioteca.java
import java.util.ArrayList;

public class Biblioteca {
    private ArrayList<Livro> livros = new ArrayList<>();
    // Método para adicionar um livro à biblioteca
    public void adicionarLivro(Livro livro) {
        livros.add(livro);
    }
    // Método para exibir todos os livros na biblioteca
    public void exibirLivros() {
        System.out.println("Livros na Biblioteca:");
        for (Livro livro : livros) {
            livro.exibirInfo();
        }
    }
    // Método para remover um livro pelo título
    public void removerLivro(String titulo) {
        // Percorre a lista de livros para encontrar o livro com o título correspondente
        for (int i = 0; i < livros.size(); i++) {
            if (livros.get(i).getTitulo().equalsIgnoreCase(titulo)) {
                livros.remove(i);  // Remove o livro encontrado
                System.out.println("\nLivro \"" + titulo + "\" removido da biblioteca.\n");
                return;  // Encerra o método após remover
            }
        }
        System.out.println("\nLivro \"" + titulo + "\" não encontrado na biblioteca.\n");
    }
}